enhancements/microshift/audit-log-configuration-options.md

```diff
@@ -0,0 +1,255 @@
+---
+title: audit-log-configuration-options
+authors:
+  - "@copejon"
+reviewers:
+  - "@dhellmann: MicroShift architect"
+  - "@pacevedom: MicroShift team-lead"
+  - "@jerpeter1, Edge Enablement Staff Engineer"
+approvers:
+  - "@dhellmann"
+creation-date: 2023-01-26
+last-updated: 2022-02-21
+status: informational
+tracking-link: 
+- "https://issues.redhat.com/browse/USHIFT-2196"
+api-approvers:
+- "None"
+---
+
+# Configurable Audit Logging for MicroShift
+
+## Summary
+
+Add ability for MicroShift users to configure API server audit logging policies, log rotation and retention.
+
+## Motivation
+
+MicroShift currently uses a hardcoded audit logging policy. It should be configurable in a manner similar to OpenShift to meet customer needs.
+
+### User Stories
+
+* As a MicroShift administrator, I want to configure audit logging policies so that I can control what events are logged.
+
+* As a MicroShift administrator, I want to configure the max file size and retention policy for audit logs so that I can better manage their disk usage.
+
+### Goals
+
+- Enable MicroShift administrators to manage logging policies as a set of hierarchical profiles.
+- Provide flexibility to specify log file sizes and retention policies.
+
+### Non-Goals
+
+- Custom Rules: MicroShift is a single-user system without user groups, so custom audit log rules similar to OpenShift are not 
+required and should be explicitly marked as out of scope.
+- Support OVN-K audit log configuration. OVN-K audit log policies are managed via a configMap and do not have the same flexibility a OpenShift and Kubernetes API servers.
+
+## Proposal
+
+This proposes exposing a subset of kube-apiserver audit-log settings flags. The kube-apiserver settings will enable user control over log file rotation and retention. Users may set fields in combination to define a maximum storage limit (e.g. max num files * max single file size = total storage limit). This is a critical feature for far edge devices with limited storage capacities. On such devices, logging data accumulation risks starving the host system or cluster workloads, potentially bricking the device until human intervention can be applied.  Thus, it is necessary to provide users a means of enforcing such a limit and what actions to take at that limit. Users must also be able to select which events are logged. This will be exposed as a set of "profiles," with predefined behaviours and will be mutually exclusive. This should give users holistic control of their audit log rotation, retention, and max allowable storage allocation.
+
+### Workflow Description
+
+1. Administrator edits MicroShift config file to specify desired audit logging policy profile
+2. Administrator edits MicroShift config file to specify max file size, number of files total, and max age of files for logs
+3. Administrator restarts MicroShift service to apply changes
+
+### API Extensions
```

Comment:
We can agree on the field names in the code review, but in general we're going to want to have units in the names (so something like `maxFileCount`, `maxAgeDays`, and `maxSizeMB`). If we end up changing anything, we can update this doc later.